### PR TITLE
[kueuectl] Support paging on kueue CLI list commands.

### LINKED ISSUE #2183: [kueuectl] Support paging on kueue CLI list commands
**What would you like to be added**:
Support paging on kueue CLI list commands.

**Why is this needed**:

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Support paging on kueue CLI list commands.

#### Which issue(s) this PR fixes:

Fixes #2183 

#### Does this PR introduce a user-facing change?

```release-note
CLI: Support paging on kueue CLI list commands.
```

### CODE REVIEW

File: cmd/kueuectl/app/list/list_clusterqueue.go [hunk 1]
```diff
@@ -144,35 +144,56 @@ func (o *ClusterQueueOptions) validActiveFlagOptionProvided() bool {
...
 	for {
-		cql, err := o.Client.ClusterQueues().List(ctx, metav1.ListOptions{
-			LabelSelector: o.LabelSelector,
-			FieldSelector: o.FieldSelector,
-			Limit:         o.clusterQueueListRequestLimit,
-			Continue:      continueToken,
-		})
+		noHeaders := totalCount > 0
+
+		list, err := o.Client.ClusterQueues().List(ctx, opts)
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply cmd/kueuectl/app/list/list_clusterqueue.go:160]: Yes, we have it [here](https://github.com/kubernetes-sigs/kueue/blob/064b44614990b09661e96d42536422da46a2c0ce/cmd/kueuectl/app/list/list_clusterqueue.go#L179).

File: cmd/kueuectl/app/list/list_clusterqueue.go [hunk 2]
```diff
@@ -56,20 +54,17 @@ type ClusterQueueOptions struct {
 	// Active means the cluster queue has kueue.ClusterQueueActive condition with status=metav1.ConditionTrue
 	Active []bool
 
-	Client                       kueuev1beta1.KueueV1beta1Interface
-	clusterQueueListRequestLimit int64
+	Client kueuev1beta1.KueueV1beta1Interface
 
-	PrintObj printers.ResourcePrinterFunc
+	ToPrinter func(noHeaders bool) (printers.ResourcePrinterFunc, error)
```
Comment (alculquicondor) [inline cmd/kueuectl/app/list/list_clusterqueue.go:59]: can this be a regular method?

File: cmd/kueuectl/app/list/list_clusterqueue.go [hunk 3]
```diff
@@ -110,14 +105,19 @@ func (o *ClusterQueueOptions) Complete(clientGetter util.ClientGetter, cmd *cobr
 
 	o.Client = clientset.KueueV1beta1()
 
-	if !o.PrintFlags.OutputFlagSpecified() {
-		o.PrintObj = printClusterQueueTable
-	} else {
+	o.ToPrinter = func(noHeaders bool) (printers.ResourcePrinterFunc, error) {
+		if !o.PrintFlags.OutputFlagSpecified() {
+			printer := newClusterQueueTablePrinter().
+				WithNoHeaders(noHeaders)
```
Comment (alculquicondor) [inline cmd/kueuectl/app/list/list_clusterqueue.go:111]: Avoid double negatives
```suggestion
				WithHeaders(headers)
```

File: cmd/kueuectl/app/list/list_clusterqueue.go [hunk 4]
```diff
@@ -144,35 +144,56 @@ func (o *ClusterQueueOptions) validActiveFlagOptionProvided() bool {
...
+			continue
+		}
+
+		if totalCount == 0 {
+			fmt.Fprintln(o.ErrOut, "No resources found")
 			return nil
 		}
-		continueToken = cql.Continue
+
+		if err := tabWriter.Flush(); err != nil {
```
Comment (alculquicondor) [inline cmd/kueuectl/app/list/list_clusterqueue.go:188]: what happens if the first 100 elements have rows with very few characters, and the next 100 characters have long values?
Would the output be misaligned?

File: cmd/kueuectl/app/list/list_localqueue.go
Comment (alculquicondor) [inline cmd/kueuectl/app/list/list_localqueue.go:1]: Similar questions as the implementation for ClusterQueues.

File: cmd/kueuectl/app/list/list_clusterqueue.go (hunk 4 above)
Comment (mbobrovskyi) [reply cmd/kueuectl/app/list/list_clusterqueue.go:188]: No, `TabWriter` will align.

File: cmd/kueuectl/app/list/list_clusterqueue.go (hunk 2 above)
Comment (mbobrovskyi) [reply cmd/kueuectl/app/list/list_clusterqueue.go]: Good point. Thanks.

File: cmd/kueuectl/app/list/list_localqueue.go
Comment (mbobrovskyi) [reply cmd/kueuectl/app/list/list_localqueue.go:1]: Also added tests to check it.

File: cmd/kueuectl/app/list/list_clusterqueue.go (hunk 3 above)
Comment (mbobrovskyi) [reply cmd/kueuectl/app/list/list_clusterqueue.go]: Fixed

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**